### PR TITLE
Added new setting to enable Recommendations

### PR DESCRIPTION
**Pull request by sagzy**
closes https://github.com/TryGhost/Product/issues/3772

- Recommendations are disabled by default and will get enabled when a first recommendation is added. The "recommendations_enabled" setting will be not shown explicitly in Admin, it will rather be set implicitly based on whether there is at least one recommendation
- The "recommendations_enabled" setting belongs to a new setting group called "recommendations"

[Tech Spec](https://www.notion.so/ghost/Recommentions-Tech-Spec-598f09e4109f4eef8d9e32db7245a90c?pvs=4)

### CODE REVIEW

File: ghost/core/core/server/data/schema/default-settings/default-settings.json [hunk 1]
```diff
@@ -565,5 +565,15 @@
             },
             "type": "number"
         }
+    },
+    "recommendations": {
+        "recommendations_enabled": {
+            "defaultValue": "true",
```
Comment (SimonBackx) [inline ghost/core/core/server/data/schema/default-settings/default-settings.json:571]:
```suggestion
            "defaultValue": "false",
```

File: ghost/core/core/server/data/migrations/versions/5.61/2023-08-29-09-32-32-add-recommendations-enabled-setting.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const {addSetting} = require('../../utils');
+
+module.exports = addSetting({
+    key: 'recommendations_enabled',
+    value: 'true',
```
Comment (SimonBackx) [inline ghost/core/core/server/data/migrations/versions/5.61/2023-08-29-09-32-32-add-recommendations-enabled-setting.js:5]:
```suggestion
    value: 'false',
```

**Comment by SimonBackx** [comment]
We probably also want to update `ghost/core/test/utils/fixtures/default-settings.json`, `ghost/core/test/utils/fixtures/default-settings-browser.json` and `ghost/core/core/shared/settings-cache/public.js`

**Comment by sagzy** [comment]
> We probably also want to update `ghost/core/test/utils/fixtures/default-settings.json`, `ghost/core/test/utils/fixtures/default-settings-browser.json` and `ghost/core/core/shared/settings-cache/public.js`

Thanks! Added these as well

File: ghost/core/core/server/data/migrations/versions/5.61/2023-08-29-09-32-32-add-recommendations-enabled-setting.js (hunk 2 above)
Comment (sagzy) [reply ghost/core/core/server/data/migrations/versions/5.61/2023-08-29-09-32-32-add-recommendations-enabled-setting.js]: Updated, following the Slack discussion [here](https://ghost.slack.com/archives/C04TMVA1D7A/p1693310637853019)

File: ghost/core/core/server/data/schema/default-settings/default-settings.json (hunk 1 above)
Comment (sagzy) [reply ghost/core/core/server/data/schema/default-settings/default-settings.json]: Updated, following the Slack discussion [here](https://ghost.slack.com/archives/C04TMVA1D7A/p1693310637853019)

**Review by SimonBackx — APPROVED** [review]

**Pull request merged**